config/config.go

```diff
@@ -225,6 +207,234 @@ func validateReferencesExist(c configpb.Configuration) error {
 	return mErr.ErrorOrNil()
 }
 
+// validateName validates an entity name is non-empty and contains no prefix that overlaps with a
+// TestGrid file prefix, post-normalization.
+func validateName(s string, entity string) error {
+	name := normalize(s)
+	if name == "" {
+		return &ConfigError{s, entity, "Normalized name can't be empty."}
```

Comment:
a) How are we using these strongly typed errors?
b) Why do we need to return a reference?

Reply:
Most of what I want is to easily display all the errors present in the config to the end user when validating config, and make it clear what parts of the config need to be fixed (i.e. what type + name errored). I'm not sure if these kinds of errors are best or if I should just switch to structured logging at this point?

Reply:
Refactored a bit to create ConfigErrors in fewer places.